designs/delay-node-enhancements/delay-node-enhancements.md

```diff
@@ -0,0 +1,83 @@
+---
+state: draft
+---
+
+# delay node enhancements
+## Summary
+ The enhancements for delay node change message queue handling and provide logging function.
+
+## Author
+ @kazuhitoyokoi
+
+## Details
+### Use cases
+#### 1. Incoming message control
+ Because Node-RED accepts all incoming message from input nodes like mqtt-in node, Node.js process sometimes reaches the upper limit of memory usage.
+To avoid the risk of message queue overflow, rate limit functionality in delay node will be useful but Node-RED users cannot customize queue handling.
+For example, the current delay node cannot simultaneously use the drop option and queue to process burst incoming messages.
+
+#### 2. Changing properties on demand
+ Currently, the delay node supports msg.delay to change property value on demand.
+If the delay node additionally supports msg.rate to set rate value and msg.queueLength to change queue size, users can create custom delay node using subflow to wrap delay node.
+
+#### 3. Logging and error handling
+ Because current drop option in rate limit mode doesn't record dropped messages, Node-RED users aren't aware of the lost messages.
+Therefore, it will be useful for Node-RED users to track background behaviors if the delay node outputs the number of dropped messages to the log stream.
```

Comment:
when would it output this information ? every time it dropped a message or only when it sends the next one ? One of the main uses for dropping messages is to reduce processing bandwidth / by adding logging on every message we are increasing it.

Reply:
The delay outputs log every 15 seconds when there are dropped messages after it is deployed in the runtime.